accepted/2022/simplify-rid-model.md

```diff
@@ -0,0 +1,221 @@
+
+## Source-build
+
+A major design tenet of the RID system is maximizing portability of code. This is particularly true for Linux. That makes sense from the perspective of Microsoft wanting to make one build of .NET available across many Linux distros, separately for both glibc and musl. It makes less sense for distros themselves building .NET from source and publishing binaries to a distro package feed.
+
+We sometimes refer to `linux-x64` (and equally to `linux-arm64` and `linux-x86`) as the "portable Linux RID". As mentioned in the libc section, this RID establishes a wide glibc compatibility range. In addition, the RID also establishes broad compatibility across distros (and their associated versions) for .NET dependencies, like OpenSSL and ICU. The way this particular form of compatibility shows up is observable but is nuanced (and isn't explained here).
+
+We'll use Red Hat as an example of an organization that builds .NET from source, to complete this discussion. They have raised concerns on [source-build not playing nicely with `linux-x64`](https://github.com/dotnet/runtime/pull/62942#issuecomment-1056942235).
+
+For Red Hat, it makes sense to accept and support `linux-x64` assets, but not to produce or offer them. Instead, Red Hat would want to produce `rhel-x64` assets. It's easiest to describe this design-point in terms of concrete scenarios.
+
+**NuGet packages** -- NuGet authors will typically value breadth and efficiency. For example, a package like `SkiaSharp` might target `linux-x64`, but not `rhel-x64`. There is no technical reason for that package to include distro-specific assets. Also, if the author produced a `rhel-x64` asset, they would need to consider producing an `ubuntu-x64` and other similar assets for other distros, and that's not scalable. As a result, it makes sense for Red Hat to support `linux-x64`. There is no technical downside to Red Hat doing so.
+
+**Runtime and host packs** -- The .NET SDK doesn't include all assets that are required for all scenarios, but instead relies on many assets being available on a NuGet feed (nuget.org or otherwise). That's a good design point for a variety of reasons. Runtime packs are a concrete scenario, which are used for building self-contained apps. Red Hat should be able to (A) build RHEL-specific runtime packages, (B) publish them to a feed of their choice, and (C) enable their users to download them by specifying a RHEL-specific RID as part of a typical .NET CLI workflow. RHEL-specific runtime packs would not be compatible in the same way as `linux-x64`. They would support `linux-x64` NuGet assets, but would only (presumably) support being run on Red Hat Enterprise Linux family OSes. For example, the RHEL-specific runtime for RHEL 8 would only support the default OpenSSL version that is provided in the RHEL 8 package feed.
```

Comment:
Can you include the RHEL-specific rid explicitly in this section (as it gets used in the rest of the doc)?
Something like:

```suggestion
**Runtime and host packs** -- The .NET SDK doesn't include all assets that are required for all scenarios, but instead relies on many assets being available on a NuGet feed (nuget.org or otherwise). That's a good design point for a variety of reasons. Runtime packs are a concrete scenario, which are used for building self-contained apps. Red Hat should be able to (A) build RHEL-specific runtime packages, (B) publish them to a feed of their choice, and (C) enable their users to download them by specifying a RHEL-specific RID (`rhel-x64`) as part of a typical .NET CLI workflow. RHEL-specific packs would not be compatible in the same way as `linux-x64`. They packs would (presumably) only support being run on Red Hat Enterprise Linux family OSes. For example, the RHEL-specific runtime for RHEL 8 would only support the default OpenSSL version that is provided in the RHEL 8 package feed.
```

Reply:
Your edit makes it sound like you would not support `linux-x64` NuGet assets. That doesn't match my intuition. Can you elaborate on your thinking?

Reply:
I meant to clarify that the packs built by Red Hat have the `rhel-x64` rid, and not the `linux-x64` rid.

> RHEL-specific packs would not be compatible in the same way as `linux-x64`. They packs would (presumably) only support being run on Red Hat Enterprise Linux family OSes.

Maybe this makes it more clear:

> RHEL-specific packs would not be compatible in the same way as `linux-x64` packs. The `rhel-x64` packs would (presumably) only support being run on Red Hat Enterprise Linux family OSes.

Reply:
Adding some more text:

> RHEL-specific packs would not be compatible in the same way as the `linux-x64` packs (built by Microsoft). The `rhel-x64` packs would (presumably) only support being run on Red Hat Enterprise Linux family OSes, while the `linux-x64` packs work across a range of distros.

Reply:
You are still removing my text on `linux-x64` NuGet assets. I'm talking about a third-party NuGet package. That remains relevant and needs to be explained.

How about this:

> RHEL-specific packs would not be compatible in the same way as the `linux-x64` packs (built by Microsoft). The `rhel-x64` packs would (presumably) only support being run on Red Hat Enterprise Linux family OSes, while the `linux-x64` packs work across a range of distros. However,`linux-x64` NuGet assets (in third-party NuGet packages) would be usable, compatible, and restorable with both Red Hat (distro-specific) and Microsoft (portable) builds of .NET.

Reply:
> linux-x64 NuGet assets (in third-party NuGet packages) would be usable, compatible, and restorable with both Red Hat (distro-specific) and Microsoft (portable) builds of .NET.

I think this belongs in the previous paragraph on _NuGet packages_.

Reply:
I'm sorry. You were right the whole time. Thanks for being patient.

Check my latest commit.

Reply:
https://github.com/dotnet/designs/pull/260/commits/8eb6dae68635228eacfea2dbdc0a430114267ea8

Reply:
lgtm, thanks.